mir_perception/mir_handle_detection/ros/src/drawer_handle_perceiver.cpp

```diff
@@ -214,6 +212,13 @@ void DrawerHandlePerceiver::extractPlaneOutlier(const PCloudT::Ptr &input, PClou
     this->project_inliers.setIndices(inliers);
     this->project_inliers.filter(*pc_plane);
 
+    Eigen::Vector4f normal_vector(0.0, 0.0, 0.0, 0.0);
+    bool cluster_success = this->getDrawerPlaneNormal(pc_plane, normal_vector);
+    this->pose_stamped.pose.orientation.x = normal_vector[0];
```

Comment:
I don't think this is correct. The normal still needs to be converted into a quaternion. [This](http://www.pcl-users.org/Two-questions-concerning-normal-estimation-td4043212.html) might help. Also [extractPlaneOrientations](https://borstel.etit.tu-chemnitz.de:8080/svn/tuc-ros-pkg/trunk/stacks/tuc_pelican/kinect_utils/src/corridor_extraction.cpp) could be useful (you still would need to convert the Euler angles to a quaternion)

Reply:
Done